src/allmydata/scripts/runner.py

```diff
@@ -223,21 +262,43 @@ def write_coverage_data():
     reactor.addSystemEventTrigger('after', 'shutdown', write_coverage_data)
 
 
-def _run_with_reactor(reactor):
+def _run_with_reactor(reactor, config, argv, stdout, stderr):
+    """
+    Run a Tahoe-LAFS node using the given reactor.
+
+    :param reactor: The reactor to use.  This implementation largely ignores
+        this and lets the rest of the implementation pick its own reactor.
+        Oops.
+
+    :param twisted.python.usage.Options config: The config object to use to
+        parse the argument list.
 
-    _setup_coverage(reactor)
+    :param argv: See ``run``.
 
-    d = defer.maybeDeferred(parse_or_exit_with_explanation, sys.argv[1:])
+    :param stdout: See ``run``.
+    :param stderr: See ``run``.
+
+    :return: A ``Deferred`` that fires when the run is complete.
+    """
+    _setup_coverage(reactor, argv)
```

Comment:
Why is additional setup for coverage needed here?  Could we document the reasons?

Reply:
added some more words in 9958236c3